packages/context/src/notification/NotificationProvider.tsx

```diff
@@ -0,0 +1,182 @@
+    instanceId: string;
+    documentId: string;
+    type: string;
+};
+
+export type InstanceChange = string;
+
+export type NotificationContextType = {
+    documentChanges?: Subscription<DocumentChange>;
+    instanceChanges?: Subscription<InstanceChange>;
+};
+
+export const NotificationContext = createContext<NotificationContextType>({});
+
+NotificationContext.displayName = 'NotificationContext';
+
+function NotificationProvider({ children }: { children: React.ReactNode }) {
+    const [instancesNotification, setInstancesNotification] = useState<HubConnection>();
+    const [documentsNotification, setDocumentsNotification] = useState<HubConnection>();
+
+    const api = useApiServices();
+
+    useEffect(() => {
+        const getConnectionInfo = (hubName: string) => {
+            return api.post<NotificationConnectionInfo>(`/notification/hubs/${hubName}/negotiate`);
+        };
+
+        const getConnection = async () => {
+            try {
+                const instancesConnectionInfo = await getConnectionInfo('instanceChanges');
+                const documentsConnectionInfo = await getConnectionInfo('documentChanges');
+
+                if (instancesConnectionInfo) {
+                    const options = {
+                        accessTokenFactory: async () => {
+                            if (instancesConnectionInfo.accessToken) {
+                                return instancesConnectionInfo.accessToken;
+                            } else {
+                                return getConnection();
+                            }
+                        },
+                    };
+
+                    const connection = new HubConnectionBuilder()
+                        .withUrl(instancesConnectionInfo.url, options as unknown as HttpTransportType)
+                        .configureLogging(LogLevel.Error)
+                        .withAutomaticReconnect()
+                        .build();
+
+                    setInstancesNotification(connection);
+                }
+
+                if (documentsConnectionInfo) {
+                    const options = {
+                        accessTokenFactory: async () => {
+                            if (documentsConnectionInfo.accessToken) {
+                                return documentsConnectionInfo.accessToken;
+                            } else {
+                                return getConnection();
+                            }
+                        },
+                    };
+
+                    const connection = new HubConnectionBuilder()
+                        .withUrl(documentsConnectionInfo.url, options as unknown as HttpTransportType)
+                        .configureLogging(LogLevel.Error)
+                        .withAutomaticReconnect()
+                        .build();
+
+                    setDocumentsNotification(connection);
+                }
+                // eslint-disable-next-line no-empty
+            } catch (err) {}
```

Comment:
Should at least log the error

Reply:
From an api standpoint, yeah, there's no reason to tie ourselves to signalr, but I think it's fine for diagnostic logging.